app/ts/library/asyncState.ts

```diff
@@ -30,7 +30,7 @@ export function useAsyncState<T>(): AsyncState<T> {
 			const resolvedState = { state: 'resolved' as const, value: resolvedValue }
 			setCapturedResult(resolvedState)
 		} catch (unknownError: unknown) {
-			const error = unknownError instanceof Error ? unknownError : typeof unknownError === 'string' ? new Error(unknownError) : new Error(`Unknown error occurred.\n${JSON.stringify(unknownError)}`)
+			const error = unknownError as E
```

Comment:
you should verify this is correct type before casting it to E. in javascript you can throw anything